### ISSUE TITLE
Mixed AO/heap partitioned table missing rows after COPY FROM 

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#6678]
On GPDB master:

```
create table mixed_ao_part(distkey int, partkey int)
with (appendonly=true) distributed by(distkey)
partition by range(partkey) (
  partition p1 start(0) end(100) with (appendonly = false),
   partition p2 start(100) end(199)
);
copy mixed_ao_part from stdin;
1	95
2	96
3	97
4	98
5	99
1	100
2	101
3	102
4	103
5	104
\.

-- Half of the rows are missing
select * from mixed_ao_part;
 distkey | partkey 
---------+---------
       4 |     103
       2 |     101
       5 |     104
       1 |     100
       3 |     102
(5 rows)
```

**Comment by Eulerizeit** [comment]
Do you know when the error occurs? Is it the copy function that is somehow
broken?  Do you know if this is broken in 5.x?

-- Rob

**Comment by hlinnaka** [comment]
It works in GPDB 5. The bug was related to GPDB changes to new code that came with the 9.2 merge.

**Comment by hlinnaka** [comment]
Fixed with commit cfcda0bad3.
